src/draw.c

```diff
@@ -281,6 +281,13 @@ static struct dimensions calculate_dimensions(GSList *layouts)
 
         dim.w += 2 * settings.frame_width;
         dim.corner_radius = MIN(dim.corner_radius, dim.h/2);
+
+        /* clamp max width to screen width */
+        const struct screen_info *scr = output->get_active_screen();
+        if (dim.w > scr->w) {
```

Comment:
When an offset is used, the notification is still partly off screen.
```suggestion
        if ((dim.w + settings.offset.x) > scr->w) {
```

Reply:
Ah yes, did't had the offset in mind.
Originally it was:
```
if (total_width > scr->w) {
    dim.w = scr->w - settings.geometry.x * 2;
}
```
which would match the change you suggested apart from the offset doubling.

I guess the offset is applied times two because of centering in the original solution?

Reply:
> I guess the offset is applied times two because of centering in the original solution?

Oh yeah that may be. That reminds me , when the notification is horizontally centered, the offset should be subtracted twice.

Reply:
I've tested again with offsets and we only need to apply the offset once in the calculation as the centering is determined elsewhere in the layout code.
I added an additional commit to this PR which calculates the correct width correctly considering the offset.